### DISCUSSION TITLE
CDlatex in Spacemacs

### DISCUSSION THREAD

**Discussion by RaghuNaik** [syl20bnr/spacemacs#16365] [General]
I am not sure how to setup **cdlatex** in org mode in Spacemacs. Right now I have `  (add-hook 'org-mode-hook #'turn-on-org-cdlatex)` under by user config section.  Not sure if this is enough because the `tab` key is not functioning as expected:

1. Pressing the tab key does not get me out of subscript / superscript braces, so that I can continue typing on the same line. 
2. Instead, it just takes to the beginning of the next line

I was wondering if there are any conflicts with evil-mode that requires a more specific configuration, say, in the presence of the latex layer. 

Would appreciate any pointers. 

Cheers.

**Comment by fnussbaum** [comment]
I cannot reproduce your issue, it seems to work correctly for me. Could you provide concrete reproduction steps as well as your system info?

For reference, this is the output of `SPC h d s` for me:
#### System Info :computer:
- OS: gnu/linux
- Emacs: 29.3
- Spacemacs: 0.999.0
- Spacemacs branch: develop (rev. a58a7d79b)
- Graphic display: t
- Running in daemon: t
- Distribution: spacemacs-base
- Editing style: vim
- Completion: compleseus
- Layers:
```elisp
(ivy helm
     (compleseus :variables vertico-cycle t embark-confirm-act-all nil embark-quit-after-action
                 '((kill-buffer . t)
                   (switch-to-buffer . t)
                   (switch-to-buffer-other-window . t)
                   (compleseus-spacemacs-help//layer-action-open-dired . t)
                   (compleseus-spacemacs-help//layer-action-open-readme-edit . t)
                   (compleseus-spacemacs-help//layer-action-open-config . t)
                   (compleseus-spacemacs-help//layer-action-open-packages . t)
                   (compleseus-spacemacs-help//layer-action-open-funcs . t)
                   (compleseus-spacemacs-help//layer-action-open-layers . t)
                   (compleseus-spacemacs-help//layer-action-open-readme . t)
                   (t)))
     spacemacs-completion spacemacs-layouts spacemacs-editing spacemacs-editing-visual
     (spacemacs-evil :variables evil-cross-lines t evil-jumps-cross-buffers t)
     spacemacs-language spacemacs-misc spacemacs-modeline spacemacs-navigation spacemacs-org spacemacs-project
     (spacemacs-visual :packages
                       (not popwin))
     (auto-completion :variables auto-completion-idle-delay 0 auto-completion-minimum-prefix-length 1 auto-completion-return-key-behavior nil auto-completion-tab-key-behavior nil auto-completion-complete-with-key-sequence "jk" auto-completion-complete-with-key-sequence-delay 0.3 auto-completion-enable-snippets-in-popup t auto-completion-enable-sort-by-usage t)
     emacs-lisp
     (scala :variables sbt:prefer-nested-projects t scala-sbt-window-position 'bottom)
     c-c++
     (python :variables python-backend 'lsp python-lsp-server 'pyright)
     rust javascript org
     (latex :variables latex-backend 'company-auctex)
     bibtex html yaml markdown csv
     (git :variables magit-display-buffer-function 'magit-display-buffer-same-window-except-diff-v1 magit-log-section-commit-count 10)
     version-control kubernetes
     (shell :variables shell-default-position 'bottom shell-default-shell 'vterm vterm-max-scrollback 100000 spacemacs-vterm-history-file-location "~/.bash_history")
     (pdf :packages
          (not pdf-view-restore)
          :variables pdf-view-use-scaling
          (string-equal
           (system-name)
           "laptopf"))
     (lsp :variables lsp-use-workspace-root-for-server-default-directory t lsp-enable-on-type-formatting nil lsp-ui-sideline-enable nil lsp-headerline-breadcrumb-enable nil)
     ibuffer neotree evil-better-jumper multiple-cursors helpful syntax-checking spell-checking colors fn-layouts fn-navigation fn-org fn-calendar fn-dired fn-gptel fn-viper fn-popper)
```
- System configuration features: ACL CAIRO DBUS FREETYPE GIF GLIB GMP GNUTLS GSETTINGS HARFBUZZ IMAGEMAGICK JPEG JSON LCMS2 LIBSELINUX LIBXML2 MODULES NATIVE_COMP NOTIFY INOTIFY PDUMPER PNG RSVG SECCOMP SOUND SQLITE3 THREADS TIFF TOOLKIT_SCROLL_BARS TREE_SITTER X11 XDBE XIM XINPUT2 XPM GTK3 ZLIB

**Reply by RaghuNaik** [reply]
Thanks for your reply. 

Disabling org-pretty-entities (,T,e) made the issue go away. Not sure if there is a way to disable this automatically when one is using CDLaTeX to write an equation in org mode. 

The requested system configuration is as below. 

Regards.

``
#### System Info :computer:
- OS: gnu/linux
- Emacs: 30.0.50
- Spacemacs: 0.999.0
- Spacemacs branch: develop (rev. a58a7d79b)
- Graphic display: t
- Running in daemon: nil
- Distribution: spacemacs
- Editing style: vim
- Completion: compleseus
- Layers:
```elisp
(html
 (auto-completion :variables auto-completion-enable-help-tooltip t
                  auto-completion-enable-snippets-in-popup t
                  auto-completion-enable-sort-by-usage t)
 bm bibtex compleseus colors emacs-lisp epub
 (ess :variables ess-enable-smart-equals t ess-enable-electric-spacing-r t
      ess-r-backend 'lsp ess-use-flymake nil ess-enable-smartparens t)
 eww git latex lsp markdown multiple-cursors
 (org :variables org-enable-modern-support t org-enable-appear-support t
      org-enable-sticky-header t org-enable-reveal-js-support t
      org-enable-asciidoc-support t org-want-todo-bindings t)
 pandoc
 (plantuml :variables plantuml-jar-path "~/plantuml.jar" org-plantuml-jar-path
           "~/plantuml.jar" plantuml-default-exec-mode 'library)
 search-engine
 (shell :variables shell-default-height 30 shell-default-position 'bottom)
 (spell-checking :variables spell-checking-enable-auto-dictionary t)
 syntax-checking spacemacs-completion spacemacs-defaults spacemacs-editing
 spacemacs-editing-visual spacemacs-evil spacemacs-visual themes-megapack
 treemacs unicode-fonts version-control)
```
- System configuration features: ACL CAIRO DBUS FREETYPE GIF GLIB GMP GNUTLS GPM GSETTINGS HARFBUZZ JPEG JSON LCMS2 LIBOTF LIBSELINUX LIBSYSTEMD LIBXML2 M17N_FLT MODULES NOTIFY INOTIFY PDUMPER PNG RSVG SECCOMP SOUND THREADS TIFF TOOLKIT_SCROLL_BARS TREE_SITTER WEBP X11 XDBE XIM XINPUT2 XPM GTK3 ZLIB

**Reply by RaghuNaik** [reply]
I think I will have to fine tune org-appear setup that will nullify org-pretty-entities in evil insert mode. 

R.

**Reply by fnussbaum** [reply]
Thanks, I can now reproduce the issue. Try simply setting `org-appear-inside-latex` to `t`.

**Reply by RaghuNaik** [reply]
Thanks. 

This conversation https://emacs-orgmode.gnu.narkive.com/Kgn9ytGT/o-bug-problems-with-sub-super-script-when-using-org-cdlatex-org-pretty-entities-org-catch-invisible- also provides a solution how not to apply `org-pretty-entities` to subscripts and superscripts, thereby making it easier to leverage cdlatex tabs. 

Was wondering if cdlatex configuration can be added to the org layer to make this fine tuning easier. 

Cheers.
